### PR TITLE
Add  multi-framework dlls to nuget packages

### PR DESCRIPTION
**Pull request by yiylin**

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/CreatePackage.cmd [hunk 1]
```diff
@@ -11,10 +11,10 @@ for /f "usebackq tokens=*" %%i in (`..\packages\vswhere.2.2.7\tools\vswhere -lat
 if not exist ..\nuget mkdir ..\nuget
 if exist ..\nuget\Microsoft.Recognizers.Text.DateTime*nupkg erase /s ..\nuget\Microsoft.Recognizers.Text.DateTime*nupkg
 "%MSBuildDir%\MSBuild\15.0\Bin\MSBuild.exe" /property:Configuration=release Microsoft.Recognizers.Text.DateTime.csproj
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.dll).FileVersionInfo.FileVersion"') do set basic=%%v
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.Number.dll).FileVersionInfo.FileVersion"') do set number=%%v
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.NumberWithUnit.dll).FileVersionInfo.FileVersion"') do set numberWithUnit=%%v
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.DateTime.dll).FileVersionInfo.FileVersion"') do set version=%%v
+for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\net45\Microsoft.Recognizers.Text.dll).FileVersionInfo.FileVersion"') do set basic=%%v
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/CreatePackage.cmd]: I'd prefer if you get the version info from the net462 path. There should be no difference, but 4.6.2 is the "master" version.

File: .NET/Microsoft.Recognizers.Text.Number/CreatePackage.cmd [hunk 2]
```diff
@@ -11,8 +11,8 @@ for /f "usebackq tokens=*" %%i in (`..\packages\vswhere.2.2.7\tools\vswhere -lat
 if not exist ..\nuget mkdir ..\nuget
 if exist ..\nuget\Microsoft.Recognizers.Text.Number*.nupkg erase /s ..\nuget\Microsoft.Recognizers.Text.Number*.nupkg
 "%MSBuildDir%\MSBuild\15.0\Bin\MSBuild.exe" /property:Configuration=release Microsoft.Recognizers.Text.Number.csproj
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.dll).FileVersionInfo.FileVersion"') do set basic=%%v
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.Number.dll).FileVersionInfo.FileVersion"') do set numberVersion=%%v
+for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\net45\Microsoft.Recognizers.Text.dll).FileVersionInfo.FileVersion"') do set basic=%%v
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Number/CreatePackage.cmd]: Same

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/CreatePackage.cmd [hunk 3]
```diff
@@ -11,9 +11,9 @@ for /f "usebackq tokens=*" %%i in (`..\packages\vswhere.2.2.7\tools\vswhere -lat
 if not exist ..\nuget mkdir ..\nuget
 if exist ..\nuget\Microsoft.Recognizers.Text.NumberWithUnit*nupkg erase /s ..\nuget\Microsoft.Recognizers.Text.NumberWithUnit*nupkg
 "%MSBuildDir%\MSBuild\15.0\Bin\MSBuild.exe" /property:Configuration=release Microsoft.Recognizers.Text.NumberWithUnit.csproj
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.dll).FileVersionInfo.FileVersion"') do set basic=%%v
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.Number.dll).FileVersionInfo.FileVersion"') do set number=%%v
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.NumberWithUnit.dll).FileVersionInfo.FileVersion"') do set version=%%v
+for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\net45\Microsoft.Recognizers.Text.dll).FileVersionInfo.FileVersion"') do set basic=%%v
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.NumberWithUnit/CreatePackage.cmd]: Same

File: .NET/Microsoft.Recognizers.Text.DateTime/Microsoft.Recognizers.Text.DateTime.nuspec [hunk 4]
```diff
@@ -18,8 +18,12 @@
     </dependencies>
   </metadata>
   <files>
-    <file src="..\build\package\Microsoft.Recognizers.Text.DateTime.dll" target="lib\net452"/>
-    <file src="..\build\package\Microsoft.Recognizers.Text.DateTime.pdb" target="lib\net452" />
+    <file src="..\build\package\net45\Microsoft.Recognizers.Text.DateTime.dll" target="lib\net45"/>
+    <file src="..\build\package\net45\Microsoft.Recognizers.Text.DateTime.pdb" target="lib\net45" />
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Microsoft.Recognizers.Text.DateTime.nuspec]: Let's remove the pdbs from the release packages. In the near future we'll publish symbol packages with them

File: .NET/Microsoft.Recognizers.Text.Number/Microsoft.Recognizers.Text.Number.nuspec [hunk 5]
```diff
@@ -16,8 +16,12 @@
     </dependencies>
   </metadata>
   <files>
-    <file src="..\build\package\Microsoft.Recognizers.Text.Number.dll" target="lib\net452"/>
-    <file src="..\build\package\Microsoft.Recognizers.Text.Number.pdb" target="lib\net452" />
+    <file src="..\build\package\net45\Microsoft.Recognizers.Text.Number.dll" target="lib\net45"/>
+    <file src="..\build\package\net45\Microsoft.Recognizers.Text.Number.pdb" target="lib\net45" />
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Number/Microsoft.Recognizers.Text.Number.nuspec]: Same (pdbs).

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/Microsoft.Recognizers.Text.NumberWithUnit.nuspec [hunk 6]
```diff
@@ -17,8 +17,12 @@
     </dependencies>
   </metadata>
   <files>
-    <file src="..\build\package\Microsoft.Recognizers.Text.NumberWithUnit.dll" target="lib\net452"/>
-    <file src="..\build\package\Microsoft.Recognizers.Text.NumberWithUnit.pdb" target="lib\net452" />
+    <file src="..\build\package\net45\Microsoft.Recognizers.Text.NumberWithUnit.dll" target="lib\net45"/>
+    <file src="..\build\package\net45\Microsoft.Recognizers.Text.NumberWithUnit.pdb" target="lib\net45" />
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.NumberWithUnit/Microsoft.Recognizers.Text.NumberWithUnit.nuspec]: Same (pdbs).

File: .NET/Microsoft.Recognizers.Text/CreatePackage.cmd [hunk 7]
```diff
@@ -11,7 +11,7 @@ for /f "usebackq tokens=*" %%i in (`..\packages\vswhere.2.2.7\tools\vswhere -lat
 if not exist ..\nuget mkdir ..\nuget
 if exist ..\nuget\Microsoft.Recognizers.Text*.nupkg erase /s ..\nuget\Microsoft.Recognizers.Text*.nupkg
 "%MSBuildDir%\MSBuild\15.0\Bin\MSBuild.exe" /property:Configuration=release Microsoft.Recognizers.Text.csproj
-for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\Microsoft.Recognizers.Text.dll).FileVersionInfo.FileVersion"') do set basicVersion=%%v
+for /f %%v in ('powershell -noprofile "(Get-Command .\bin\release\net45\Microsoft.Recognizers.Text.dll).FileVersionInfo.FileVersion"') do set basicVersion=%%v
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text/CreatePackage.cmd]: Same

File: .NET/Microsoft.Recognizers.Text/Microsoft.Recognizers.Text.csproj [hunk 8]
```diff
@@ -14,6 +14,7 @@
 
   <ItemGroup>
     <PackageReference Include="vswhere" Version="2.2.11" />
+	<PackageReference Include="NuGet.CommandLine" Version="4.3.0" />
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text/Microsoft.Recognizers.Text.csproj]: If PR#182 merges first, don't modify this file.

File: .NET/Microsoft.Recognizers.Text/Microsoft.Recognizers.Text.nuspec [hunk 9]
```diff
@@ -15,10 +15,18 @@
     </dependencies>
   </metadata>
   <files>
-    <file src="..\build\package\Microsoft.Recognizers.Text.dll" target="lib\net452"/>
-    <file src="..\build\package\Microsoft.Recognizers.Text.pdb" target="lib\net452" />
-    <file src="..\build\package\Microsoft.Recognizers.Definitions.dll" target="lib\net452"/>
-    <file src="..\build\package\Microsoft.Recognizers.Definitions.pdb" target="lib\net452" />
+    <file src="..\build\package\net45\Microsoft.Recognizers.Text.dll" target="lib\net45"/>
+    <file src="..\build\package\net45\Microsoft.Recognizers.Text.pdb" target="lib\net45" />
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text/Microsoft.Recognizers.Text.nuspec]: Same (pdbs).

File: .NET/Samples/BotBuilder/BotBuilderRecognizerBot.csproj [hunk 10]
```diff
@@ -72,9 +72,8 @@
       <HintPath>..\..\packages\Microsoft.WindowsAzure.ConfigurationManager.3.2.3\lib\net40\Microsoft.WindowsAzure.Configuration.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
-      <Private>True</Private>
+    <Reference Include="Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\Newtonsoft.Json.10.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (tellarin) [inline .NET/Samples/BotBuilder/BotBuilderRecognizerBot.csproj]: I like to keep json dependencies at the newest, but did you test it the bot still works fine?

File: .NET/Samples/SimpleConsole/packages.config [hunk 11]
```diff
@@ -1,4 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Newtonsoft.Json" version="10.0.3" targetFramework="net461" />
+  <package id="NuGet.CommandLine" version="4.3.0" targetFramework="net462" developmentDependency="true" />
```
Comment (tellarin) [inline .NET/Samples/SimpleConsole/packages.config]: Perhaps we should remove this as not really required here. Devs reading the sample may get the wrong impression.

Comment (yiylin) [reply .NET/Samples/SimpleConsole/packages.config]: Ok. I will remove

File: .NET/Samples/BotBuilder/BotBuilderRecognizerBot.csproj (hunk 10 above)
Comment (yiylin) [reply .NET/Samples/BotBuilder/BotBuilderRecognizerBot.csproj]: No. I will test it tomorrow

File: .NET/Samples/SimpleConsole/packages.config (hunk 11 above)
Comment (yiylin) [reply .NET/Samples/SimpleConsole/packages.config]: If I remove this, then I can not pass the build because of the missing Nuget.CommandLine

Comment (yiylin) [reply .NET/Samples/SimpleConsole/packages.config]: Even this System.Collections.Immutable may not be downloaded. Maybe we create a tool/package project to reference these packages.

**Pull request merged**